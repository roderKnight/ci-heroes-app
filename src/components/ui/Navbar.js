import React from 'react'
import { Link, NavLink, useNavigate } from 'react-router-dom'
import { SearchForm } from '../search/SearchForm';

export const Navbar = () => {

    const navigate = useNavigate();

    const handleLogout = () => {
        navigate('login', {
            replace: true
        });
    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark px-5 py-2">
            
            <Link 
                className={`${isActive => (!isActive ? "text-muted" : "")} navbar-brand`} 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        className={`${isActive => (!isActive ? "" : "active")} nav-item nav-link`}
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        className={`${isActive => (!isActive ? "" : "active")} nav-item nav-link`}
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <SearchForm />
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                <ul className="navbar-nav ml-auto">

                    <span className="nav-item nav-link text-info">
                        Rodrigo
                    </span>

                    <button 
                        className = "nav-item nav-link btn"
                        onClick = { handleLogout }
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}
