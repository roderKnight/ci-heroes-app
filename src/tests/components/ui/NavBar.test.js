import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('tests on <NavBar />', () => {
   
    test('should the NavBar component be renderized properly', () => {
        const contextValue = {
            user: {
                logged: true,
                name: 'Pedro'
            }
        };

        const wrapper = mount( 
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['']} value={{contextValue}}>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim()).toBe('Pedro');
    });


    test('should call to logout, call navigate and dispatch with their arguments', () => {
        const contextValue = {
            dispatch: jest.fn(), // mock of our dispatch
            user: {
                logged: true,
                name: 'Pedro'
            }
        };

        const wrapper = mount( 
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['']} value={{contextValue}}>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        wrapper.find({className: "nav-item nav-link btn"}).prop('onClick')();

        expect( mockNavigate ).toHaveBeenCalledWith('login', {replace: true});  
        expect( contextValue.dispatch ).toHaveBeenCalledWith({'type': types.logout});
        //expect( wrapper.find({className: "nav-item nav-link btn"}).text().trim()).toBe('Logout');
    });
    
    
});
