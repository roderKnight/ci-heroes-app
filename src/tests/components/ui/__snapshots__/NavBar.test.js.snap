// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <NavBar /> should the NavBar component be renderized properly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark px-5 py-2"
>
  <a
    className="isActive => !isActive ? \\"text-muted\\" : \\"\\" navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="isActive => !isActive ? \\"\\" : \\"active\\" nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="isActive => !isActive ? \\"\\" : \\"active\\" nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <form
        className="d-flex mx-5"
        onSubmit={[Function]}
      >
        <input
          aria-label="Search"
          autoComplete="off"
          className="searc-navbar form-control me-2"
          name="searchText"
          onChange={[Function]}
          placeholder="Search"
          type="search"
          value=""
        />
        <button
          className="btn btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pedro
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
